adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -119,6 +118,480 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
   def transform(tFn: RDD[T] => RDD[T]): U = {
     replaceRdd(tFn(rdd))
   }
+  val self = this
+
+  /**
+   *
+   */
+  private[rdd] object SortedTrait extends Serializable {
+    var isSorted: Boolean = false
```

Comment:
This is my personal bias, but I think we gain more from this being immutable than from it being a singleton object.